bindings/go/configuration/filesystem/v1alpha1/config.go

```diff
@@ -0,0 +1,105 @@
+package v1alpha1
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	v1 "ocm.software/open-component-model/bindings/go/configuration/v1"
+	"ocm.software/open-component-model/bindings/go/runtime"
+)
+
+const (
+	// ConfigType defines the type identifier for credential configurations
+	ConfigType = "filesystem.config.ocm.software"
+)
+
+var scheme = runtime.NewScheme()
+
+func init() {
+	scheme.MustRegisterWithAlias(&Config{}, runtime.NewVersionedType(ConfigType, Version))
+}
+
+// Config represents the top-level configuration for the plugin manager.
+//
+// +k8s:deepcopy-gen:interfaces=ocm.software/open-component-model/bindings/go/runtime.Typed
+// +k8s:deepcopy-gen=true
+// +ocm:typegen=true
+type Config struct {
+	Type runtime.Type `json:"-"`
+	// TempFolder defines places where plugins and other functionalities can put ephemeral files under.
+	TempFolder string `json:"tempFolder,omitempty"`
+}
+
+type Duration time.Duration
+
+func (d Duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(d).String())
+}
+
+func (d *Duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+
+	return json.Unmarshal(b, &v)
+}
+
+// LookupConfig creates a new filesystem configuration from a central V1 config.
+func LookupConfig(cfg *v1.Config) (*Config, error) {
+	var merged *Config
+	if cfg != nil {
+		cfg, err := v1.Filter(cfg, &v1.FilterOptions{
+			ConfigTypes: []runtime.Type{
+				runtime.NewVersionedType(ConfigType, Version),
+				runtime.NewUnversionedType(ConfigType),
+			},
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to filter config: %w", err)
+		}
+		cfgs := make([]*Config, 0, len(cfg.Configurations))
+		for _, entry := range cfg.Configurations {
+			var config Config
+			if err := scheme.Convert(entry, &config); err != nil {
+				return nil, fmt.Errorf("failed to decode credential config: %w", err)
+			}
+			cfgs = append(cfgs, &config)
+		}
+		merged = Merge(cfgs...)
+		if merged == nil {
+			merged = &Config{}
+		}
+	} else {
+		merged = new(Config)
+	}
+
+	// TODO: Question: Should we do this, or leave it empty signaling that the using code can create their own location?
```

Comment:
I think we want plugins to be able to choose os.TempDir() right?

Reply:
Yes. So I think we shouldn't default this then. Otherwise, it will always have a value that will point to os.TempDir if not defined explicitly.